plugins/mimecast_v2/icon_mimecast_v2/tasks/monitor_siem_logs/task.py

```diff
@@ -109,24 +142,37 @@ def update_state(self, state: Dict) -> Dict:
                     state[QUERY_CONFIG][log_type] = copy.deepcopy(initial_log_type_config)
         return state
 
-    def get_max_lookback_date(self, now_date: datetime, run_condition: str, custom_config: bool) -> datetime:
+    def get_max_lookback_date(
+        self,
+        now_date: datetime,
+        run_condition: str,
+        custom_config: bool,
+        last_query_date: datetime,
+    ) -> datetime:
         """
         Get max lookback date for run condition
         :param now_date:
         :param run_condition:
         :param custom_config:
+        :param last_query_date:
         :return: max_run_lookback_date
         """
         max_run_lookback_days = MAX_LOOKBACK_DAYS
-        if run_condition in [INITIAL_RUN] and not custom_config:
+        if run_condition == INITIAL_RUN and not custom_config:
             max_run_lookback_days = INITIAL_MAX_LOOKBACK_DAYS
-
         max_run_lookback_date = now_date - timedelta(days=max_run_lookback_days)
+        if last_query_date:
+            if run_condition == PAGINATION_RUN and last_query_date == now_date - timedelta(
+                days=(max_run_lookback_days + HEADROOM_DAYS)
+            ):
+                self.logger.info("TASK: Pagination run detected, allow completion of cutoff day logs retrieval.")
+                max_run_lookback_date -= timedelta(days=HEADROOM_DAYS)
```

Comment:
I think I'm getting mixed with `lookback` and `cutoff` 😂 I mean, the headroom to allow pagination is good approach, but wonder if we couldn't freeze that `cutoff` time during pagination? 

Like the main issue is related to the `cutoff` time that's slightly getting moved forward on next run when that `cutoff` boundary is reached, right? 😃 Why we couldn't just freeze that at that point during pagination, so after it finishes paginating it bumps the time forward to `present`? 😃

Reply:
I think it should be ok as our cutoff works in full days. So we should only have an issue with cutoff if we're paginating for the 1st of the month and todays date moves over to the 8th at 12 midnight. If that happens now we just allow an extra day instead of leaving those logs behind.

I didn't want to freeze it because we'd still have to determine a maximum allowance past the cutoff for pagination and the minimum we can use is 24 hours which makes it easy to just apply a timedelta of a day if we are still within 7-8 days. Otherwise we'd still have to do a date check or else stopped and restarted integrations could continue from where they're left off instead of moving towards the cutoff.

`now_date = datetime.now(tz=timezone.utc).date()`